emop-maven-plugin/src/main/java/edu/cornell/RppHandlerMojo.java

```diff
@@ -0,0 +1,167 @@
+        if (specsToMonitor.isEmpty()) {
+            return "";
+        } else {
+            File javamopAgentFile = new File(javamopAgent);
+            File createdJar = new File(metaInfoDirectory, mode + "-javamop.jar");
+            try {
+                Files.copy(javamopAgentFile.toPath(), createdJar.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                Util.generateNewMonitorFile(
+                        metaInfoDirectory + File.separator + mode + "-ajc.xml", specsToMonitor);
+                Util.replaceFileInJar(createdJar.getAbsolutePath(), "/META-INF/aop-ajc.xml",
+                        metaInfoDirectory + File.separator + mode + "-ajc.xml");
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            } catch (MojoExecutionException ex) {
+                throw new RuntimeException(ex);
+            }
+            return createdJar.getAbsolutePath();
+        }
+    }
+
+    /**
+     * Creates new agent JARs for running critical and background phases, and sets up System properties to record
+     * paths to critical and background phase JARs.
+     */
+    private void setupJars() {
+        if (javamopAgent == null) {
+            javamopAgent = getLocalRepository().getBasedir() + File.separator + "javamop-agent"
+                    + File.separator + "javamop-agent"
+                    + File.separator + "1.0"
+                    + File.separator + "javamop-agent-1.0.jar";
+        }
+        System.setProperty("previous-javamop-agent", javamopAgent);
+        setupSpecFiles();
+        computeSpecSets();
+        String criticalRunAgentPath = setUpSingleJar("critical", criticalSpecsSet);
+        String backgroundRunAgentPath = setUpSingleJar("background", backgroundSpecsSet);
+        if (!criticalRunAgentPath.isEmpty()) {
+            System.setProperty("rpp-agent", criticalRunAgentPath);
+            System.setProperty("background-agent", backgroundRunAgentPath);
+        } else {
+            getLog().info("Critical phase had no specs, skipping and running background phase...");
+            System.setProperty("rpp-agent", backgroundRunAgentPath);
+            System.setProperty("background-agent", ""); // prevent RppMojo from running a second time
+        }
+    }
+
+    /**
+     * This mojo performs setup for RPP, and configures surefire to monitor the set of critical specs.
+     * @throws MojoExecutionException where instrumentation for surefire did not succeed.
+     */
+    public void execute() throws MojoExecutionException {
+        metaInfoDirectory = new File(getArtifactsDir());
+        // prepare the two jars
+        setupJars();
+        // load agent that will harness surefire and manipulate arguments to surefire before test execution
+        if (!AgentLoader.loadDynamicAgent("JavaAgent.class")) {
+            throw new MojoExecutionException("Could not attach agent");
+        }
+    }
+}
```

Comment:
Add new line at the end. In fact, in a separate PR, you should add a rule to checkstyle to fail if there is no new line at the end

Reply:
why is it a good practice to add a newline at the end?

Reply:
Interesting; thanks!